### DISCUSSION TITLE
I might have to wait on mariadb arm, any thoughts team?

### DISCUSSION THREAD

**Discussion by siliconplains44** [Homebrew/discussions#530] [Everyday usage]
I bought a new macbook air m1 based system and love it.  Yet I am finding a few things interesting in my journey with the new machine.  Previous to my m1 I had obviously all intel based system and homebrew has been a total champ.

I have some python stuff I need to run and so I went down the path of trying it on my new m1 based system.  A dependency of my python is mariadb.  So I tried to install mariadb on my m1 based arch arm brew installation.  The system responds with there is no bottle and then I found it was listed as missing so far.  I did find mariadb and mysql had been ported to arm though.  

Anyway, I then went and tried to install homebrew x64 and run under rosetta.  Then I tried installing mariadb and bam, error.  So I am kind fo stuck.  

arch -x86_64 brew install mariadb                                                                             
Error: Cannot install in Homebrew under Rosetta 2 in ARM default prefix (/opt/homebrew)!

Is there any workaround for this issue?

**Comment by SMillerDev** [comment]
Install a brew instance using rosetta and use that for anything that doesn't support ARM yet.

**Comment by siliconplains44** [comment]
...SO I ran the arch64 brew on mariadb and I receive an error.  What did I do wrong?  

I did this...

arch -x86_64 brew install mariadb

.. and received this...

Error: Cannot install in Homebrew under Rosetta 2 in ARM default prefix (/opt/homebrew)!

**Reply by SMillerDev** [reply]
You need a separate brew install for rosetta software. `arch -x86_64 <install script>`
